### DISCUSSION TITLE
Call elements of simple tab workflow via API?

### DISCUSSION THREAD

**Discussion by gymnae** [mcmonkeyprojects/SwarmUI#282] [Q&A] [answered]
I like the simple tab and how I can generate a simplified workflow to generate images. My users would like to access the simple tab on mobile devices, but currently the design of SwarmUI does not lend itself for mobile usage

I wondered if there are APIs for elements of a workflow generated with SwarmUI nodes for the simple tab.

For example:
1. I have a workflow where all a user can do is decide on height and width of an image and then type in their prompt. 
2. I have an extended workflow, where users have to choose a lora in addition.

Would it be possible to reach these elements via the SwarmUI API, so I could build a basic responsive HTML page?

**Comment by southbayjay** [comment]
I believe you can use the “save workflow” button and then when your asked for the name and description there’s a checkbox for “make available in simple mode” or something like that.

**Comment by gymnae** [comment]
Thanks, that is how to add the workflow wo the simple mode. What I want to achieve to be able to use workflows in the simple mode on a mobile device. In the current state, SwarmUI is not usable on a smartphone.

Therefore, I wondered if there is an API for making calls to the simple mode. I read that there are APIs for the generate mode.

**Comment by mcmonkey4eva** [comment, marked as answer]
Everything always has APIs in Swarm. Swarm's interfaces uses its own API, so anything you can do in the interface you can do via API. Just look at network traffic (via browser console on desktop) to see how to do things.
Simple mode's API isn't incredible atm (iirc it just includes the entire workflow data in the gen request), but it works.

If you're good with mobile UI dev, PRs to improve Swarm's mobile-friendliness are very welcome (as long as they don't break format on desktop devices)

**Reply by gymnae** [reply]
Thank you, I will check the output. Sadly, I can't code for shit, so you really don't want me to create a PR ;)
